lib/metasploit/framework/login_scanner/zabbix.rb

```diff
@@ -113,17 +127,16 @@ def try_credential(credential)
         #   * :proof [String] the HTTP response body
         def try_login(credential)
           res = try_credential(credential)
+
           if res && res.code == 302
-            opts = {
-              'uri'     => normalize_uri('profile.php'),
-              'method'  => 'GET',
-              'headers' => {
-                'Cookie'  => "zbx_sessionid=#{self.zsession}"
-              }
-            }
-            res = send_request(opts)
-            if (res && res.code == 200 && res.body.to_s =~ /<title>Zabbix .*: User profile<\/title>/)
+            res = perform_login_attempt('profile.php') # profile.php exists in Zabbix versions up to Zabbix 5.x
+            if (res && res.code == 200 && res.body.to_s =~ /<title>.*: User profile<\/title>/)
               return {:status => Metasploit::Model::Login::Status::SUCCESSFUL, :proof => res.body}
+            else
+              res = perform_login_attempt('/zabbix.php?action=userprofile.edit') # On version 5.x and later of Zabbix, profile.php was replaced with /zabbix.php?action=userprofile.edit
+              if (res && res.code == 200 && res.body.to_s =~ /<title>.*: User profile<\/title>/)
+                return {:status => Metasploit::Model::Login::Status::SUCCESSFUL, :proof => res.body}
```

Comment:
I don't think this is using the standard `send_request` method that modules use and is instead using [this one](https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/login_scanner/http.rb#L218) which will [raise exceptions](https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/login_scanner/http.rb#L218). (Looks like the docs saying it'll return `nil` are wrong 😬 ). When a connection error is raised, you should catch it and return `UNABLE_TO_CONNECT`, this [this one](https://github.com/rapid7/metasploit-framework/blob/master/lib/metasploit/framework/login_scanner/jupyter.rb#L53-L54) does so it can be properly handled by aborting the rest of the attempts.

Reply:
@smcintyre-r7 This should now be resolved with https://github.com/rapid7/metasploit-framework/pull/14297/commits/b5060054384b1b5dbec143efda762ca846cc5e1d